src/main/java/me/karakelley/http/handlers/StaticFilesHandler.java

```diff
@@ -1,65 +1,41 @@
 package me.karakelley.http.handlers;
 
-import me.karakelley.http.contentpresenter.ContentPresenter;
-import me.karakelley.http.filesystem.PublicDirectory;
+import me.karakelley.http.HttpMethod;
 import me.karakelley.http.Request;
 import me.karakelley.http.Response;
-import me.karakelley.http.responses.NotFound;
-import me.karakelley.http.responses.Ok;
+import me.karakelley.http.contentpresenter.ContentPresenter;
+import me.karakelley.http.filesystem.PublicDirectory;
+import me.karakelley.http.handlers.staticfilestrategies.DeleteResourceStrategy;
+import me.karakelley.http.handlers.staticfilestrategies.NewResourceStrategy;
+import me.karakelley.http.handlers.staticfilestrategies.RetrieveResourceStrategy;
+import me.karakelley.http.handlers.staticfilestrategies.UpdateResourceStrategy;
 
-import java.io.File;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 
 public class StaticFilesHandler implements Handler {
-  private final ContentPresenter contentPresenter;
-  private final PublicDirectory publicDirectory;
-  private final String CONTENT_TYPE = "Content-Type";
-  private final String TEXT_HTML = "text/html";
+  private static Map<HttpMethod, Handler> fileStrategies;
+  private static PublicDirectory publicDirectory = null;
+  private static ContentPresenter htmlPresenter = null;
 
-  public StaticFilesHandler(PublicDirectory publicDirectory, ContentPresenter contentPresenter) {
-    this.contentPresenter = contentPresenter;
+  public StaticFilesHandler(PublicDirectory publicDirectory, ContentPresenter htmlPresenter) {
     this.publicDirectory = publicDirectory;
+    this.htmlPresenter = htmlPresenter;
+    setupFileStrategies();
```

Comment:
This would re-instantiate the strategies every time the class is instantiated.

1. Changing `static` level things from the instance level is almost always a bad idea
2. This isn't any more efficient than just having `fileStrategies` be an instance variable.

Try changing this code so the `Map` is only set up one time for the life of the program.

Reply:
I don't think that's true. Once the server starts, `Application` is instantiated only once. Upon instantiation it then instantiates a new `StaticFilesHandler` which then instantiates all the strategies (just once) and it never happens again (unless you restart the server obviously).